### PR TITLE
[batch][dag 4] Require pylint and flake8 success for all PRs

### PR DESCRIPTION
**Pull request by danking**
I also added `batch/ignored-pylint-errors` which we diff the output of pylint against. If pylint differs from the `ignored-pylint-errors`, then either we fixed an ignored error or we added a new unignored error. In either case, the developer should address the issue by changing the code or the `ignored-pylint-errors` file.

### CODE REVIEW

**Comment by danking** [comment]
cc: @cseed @tpoterba I think we should head in this direction for hail/hail too.

**Comment by tpoterba** [comment]
ah nice. very nice.

I agree

File: batch/setup.cfg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[flake8]
+max-line-length=100
```
Comment (jigold) [inline batch/setup.cfg]: I thought I saw somewhere else that the maximum line length is 120. Maybe the previous PR?

File: batch/hail-ci-build.sh [hunk 2]
```diff
@@ -1,2 +1,8 @@
 #!/bin/sh
+set -ex
+
+flake8 batch
+PYLINT_OUTPUT_FILE=$(mktemp)
+pylint batch --rcfile batch/pylintrc >$PYLINT_OUTPUT_FILE --score=n || true
+diff $PYLINT_OUTPUT_FILE ignored-pylint-errors
```
[the remark this answers is not in the document]
Comment (danking) [reply batch/hail-ci-build.sh]: If there is a different between the output in `$PYLINT_OUTPUT_FILE` and `ignored-pylint-errors` `diff` will return a non-zero exit code, failing this CI build.

File: batch/ignored-pylint-errors [hunk 3]
```diff
@@ -0,0 +1,4 @@
+************* Module batch.server
```
[the remark this answers is not in the document]
Comment (danking) [reply batch/ignored-pylint-errors]: Hmm. Good point.

File: batch/setup.cfg (hunk 1 above)
Comment (danking) [reply batch/setup.cfg]: Ah good point, pylintrc and flake8 have different configuration files. I changed them both to be 120.

**Comment by jigold** [comment]
If you switch the base to master, I can approve this.

**Comment by danking** [comment]
odd, it looks like we're using python2 to test this file.

**Pull request merged**